### PR TITLE
I am creating a sample script for shadow sampler.

### PR DESCRIPTION
**Pull request by SubhajitDuttaChowdhury**
This is a simple implementation of shadow sampler.

### CODE REVIEW

File: shadow_sampler.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from torch_geometric.loader import GraphSAINTRandomWalkSampler
+from torch_geometric.loader import ShaDowKHopSampler
+from torch_geometric.nn import GraphConv
+from torch_geometric.utils import degree
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'Flickr')
+dataset = Flickr(path)
+data = dataset[0]
+row, col = data.edge_index
+data.edge_weight = 1. / degree(col, data.num_nodes)[col]  # Norm by in-degree.
```
Comment (rusty1s) [inline shadow_sampler.py:16]: I don't think we need this normalization here, as `ShaDowKHopSampler` does not perform a re-weighting.

File: shadow_sampler.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--use_normalization', action='store_true')
+args = parser.parse_args()
+
+
+indices = data.train_mask.nonzero()
+index = torch.flatten(indices)
+loader = ShaDowKHopSampler(data, depth=2, num_neighbors=5, node_idx=index)
```
Comment (rusty1s) [inline shadow_sampler.py:25]: It should work with `data.train_mask` as input as well. If not, please let me know.

File: shadow_sampler.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+
+class Net(torch.nn.Module):
```
Comment (rusty1s) [inline shadow_sampler.py:28]: Let's make direct use of `torch_geometric.nn.GraphSAGE` to reduce code size:, see https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.models.GraphSAGE.

File: shadow_sampler.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    model.set_aggr('add' if args.use_normalization else 'mean')
+
+    total_loss = total_examples = 0
+    for data in loader:
+        data = data.to(device)
+        optimizer.zero_grad()
+
+        if args.use_normalization:
+            edge_weight = data.edge_norm * data.edge_weight
+            out = model(data.x, data.edge_index, edge_weight)
```
Comment (rusty1s) [inline shadow_sampler.py:71]: It seems the model is missing the pooling step to go from subgraph-level embeddings to node-level embeddings.

**Review by rusty1s — CHANGES_REQUESTED** [review]

File: shadow_sampler.py (hunk 1 above)
Comment (SubhajitDuttaChowdhury) [reply shadow_sampler.py]: Yes, you are right. I have removed that part.

File: shadow_sampler.py (hunk 2 above)
Comment (SubhajitDuttaChowdhury) [reply shadow_sampler.py]: Yes it works. I have changed it.

File: shadow_sampler.py (hunk 4 above)
Comment (SubhajitDuttaChowdhury) [reply shadow_sampler.py]: I have added the pooling step.

File: shadow_sampler.py (hunk 3 above)
Comment (SubhajitDuttaChowdhury) [reply shadow_sampler.py:23]: Okay. I have changed it to SAGEConv.

**Review by rusty1s — APPROVED** [review]

**Comment by rusty1s** [comment]
Thank you!
